File: src/main/java/com/yupi/template/model/entity/Rating.java
package com.yupi.template.model.entity;

import com.mybatisflex.annotation.Column;
import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 用户评分实体
 *
 * @author <a href="https://codefather.cn">编程导航学习圈</a>
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("rating")
public class Rating implements Serializable {

    /**
     * 评分唯一标识
     */
    @Id
    private String id;

    /**
     * 对话ID
     */
    @Column("conversationId")
    private String conversationId;

    /**
     * 消息序号(对应某一轮对话)
     */
    @Column("messageIndex")
    private Integer messageIndex;

    /**
     * 用户ID
     */
    @Column("userId")
    private Long userId;

    /**
     * 评分类型: left_better/right_better/tie/both_bad/variant_N
     */
    @Column("ratingType")
    private String ratingType;

    /**
     * 获胜模型
     */
    @Column("winnerModel")
    private String winnerModel;

    /**
     * 失败模型
     */
    @Column("loserModel")
    private String loserModel;

    /**
     * 获胜变体索引(用于prompt_lab)
     */
    @Column("winnerVariantIndex")
    private Integer winnerVariantIndex;

    /**
     * 失败变体索引(用于prompt_lab)
     */
    @Column("loserVariantIndex")
    private Integer loserVariantIndex;

    /**
     * 创建时间
     */
    @Column("createTime")
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @Column("updateTime")
    private LocalDateTime updateTime;

    /**
     * 是否删除
     */
    @Column("isDelete")
    private Integer isDelete;

    private static final long serialVersionUID = 1L;
}

File: python-backend/app/schemas/rating.py
"""
评分相关的 Pydantic 模型
@author <a href="https://codefather.cn">编程导航学习圈</a>
"""
from typing import Optional
from pydantic import BaseModel, Field
from datetime import datetime


class RatingAddRequest(BaseModel):
    """添加或更新评分请求"""
    conversation_id: str = Field(..., description="对话ID", alias="conversationId")
    message_index: int = Field(..., description="消息序号", alias="messageIndex")
    rating_type: str = Field(..., description="评分类型: left_better/right_better/tie/both_bad/variant_N", alias="ratingType")
    winner_model: Optional[str] = Field(None, description="获胜模型", alias="winnerModel")
    loser_model: Optional[str] = Field(None, description="失败模型", alias="loserModel")
    winner_variant_index: Optional[int] = Field(None, description="获胜变体索引(用于prompt_lab)", alias="winnerVariantIndex")
    loser_variant_index: Optional[int] = Field(None, description="失败变体索引(用于prompt_lab)", alias="loserVariantIndex")
    
    model_config = {"populate_by_name": True}


class RatingVO(BaseModel):
    """评分视图对象"""
    id: str
    conversation_id: str = Field(..., alias="conversationId")
    message_index: int = Field(..., alias="messageIndex")
    user_id: int = Field(..., alias="userId")
    rating_type: str = Field(..., alias="ratingType")
    winner_model: Optional[str] = Field(None, alias="winnerModel")
    loser_model: Optional[str] = Field(None, alias="loserModel")
    winner_variant_index: Optional[int] = Field(None, alias="winnerVariantIndex")
    loser_variant_index: Optional[int] = Field(None, alias="loserVariantIndex")
    create_time: datetime = Field(..., alias="createTime")
    update_time: datetime = Field(..., alias="updateTime")

    model_config = {"from_attributes": True, "populate_by_name": True}
